cluster-test/test-orchestrator/test_yaml_parser.go

```diff
@@ -0,0 +1,308 @@
+
+	// extract size
+	size, err := strconv.Atoi(sizeStr)
+	if err != nil {
+		panic("size convert: " + err.Error())
+	}
+
+	// extract script
+	labels, cmds := extractScript(data.Script, varsData, runData)
+	script := parseScript(labels, cmds)
+
+	// extract Measure
+	var measureStrs []string
+	for _, measureStr := range data.Measure {
+		measureStrs = append(measureStrs, replace(measureStr, varsData, runData))
+	}
+	measure := parseMeasure(measureStrs)
+
+	return &Scenario{
+		Name:    name,
+		Desc:    desc,
+		Size:    size,
+		Script:  script,
+		Measure: measure,
+	}
+}
+
+func extractScript(script []map[string]string, varsData, runData []string) (labels, cmds []string) {
+	for _, cmdData := range script {
+		if len(cmdData) != 1 {
+			msg := fmt.Sprintf("\"%v\" is not a valid command, should contain exactly one entry", cmdData)
+			panic(msg)
+		}
+
+		for label, cmd := range cmdData {
+			labels = append(labels, replace(label, varsData, runData))
+			cmds = append(cmds, replace(cmd, varsData, runData))
+		}
+	}
+	return labels, cmds
+}
+
+func parseScript(labels, cmdStrs []string) []*Command {
+	defer wrapPanicf("in parse script")
+	var cmds []*Command
+	for i := range labels {
+		cmd := parseCommand(labels[i], cmdStrs[i])
+		cmds = append(cmds, cmd)
+	}
+
+	return cmds
+}
+
+func parseCommand(label, cmdString string) *Command {
+	defer wrapPanicf("in parse command '%s: %s'", label, cmdString)
+	split := split(cmdString)
+	if len(split) == 0 {
+		panic("empty command")
+	}
+
+	return &Command{
+		Label: label,
+		Cmd:   split[0],
+		Args:  split[1:],
+	}
+}
+
+func parseMeasure(msData []string) []*Measurement {
+	var ms []*Measurement
+	for _, mData := range msData {
+		ms = append(ms, parseMeasurement(mData))
+	}
+	return ms
+}
+
+func parseMeasurement(str string) *Measurement {
+	defer wrapPanicf("in parse measure '%s'", str)
+
+	split := split(str)
+	var measurementArgs = split
+
+	if len(split) < 3 {
+		panic("contains too few fields")
+	}
+
+	// search for assertion
+	var assertion *Assertion
+	for i, s := range split {
+		if s == "is" || s == "in" {
+			interval := strings.Join(split[i+1:], "")
+			assertion = parseAssertion(split[i], interval)
+			measurementArgs = split[3:i]
+		}
+	}
+
+	return &Measurement{
+		Start:     split[0],
+		End:       split[1],
+		Quantity:  split[2],
+		Args:      measurementArgs,
+		Assertion: assertion,
+	}
+}
+
+func parseAssertion(typStr string, arg string) *Assertion {
+	defer wrapPanicf("in parse assertion '%s %s'", typStr, arg)
+
+	switch typStr {
+	case "is":
+		typ := AssertionTypeIs
+		v := parseValue(arg)
+		return &Assertion{
+			Type: typ,
+			V1:   v,
+		}
+
+	case "in":
+		typ := AssertionTypeIn
+		v1, v2 := parseRange(arg)
+		return &Assertion{
+			Type: typ,
+			V1:   v1,
+			V2:   v2,
+		}
+	}
+
+	panic("not valid assertion type")
+}
+
+func parseRange(rng string) (v1, v2 Value) {
+	defer wrapPanicf("in parse range '%s'", rng)
+
+	if rng[0] != '(' || rng[len(rng)-1] != ')' {
```

Comment:
Because of the wrapPanicf, the line above it has `
```
in parse measurement 't0 t1 kill 1 in'
in parse assertion 'in '
in parse range ''
```
So I think it's clear enough then where the error lies. Do you agree?